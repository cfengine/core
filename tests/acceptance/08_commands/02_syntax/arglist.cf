#######################################################
#
# Test that arglist works properly
#
#######################################################

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent test
{
  commands:
      "$(G.echo)" arglist => { "=abc=2'3\"4" }, module => "true";
      # this will generate the command `echo =def=567 =ghi=8'9"a`
      "$(G.echo)" args => "=def=567", arglist => { "=ghi=8'9\"a" }, module => "true";
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(echo,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
