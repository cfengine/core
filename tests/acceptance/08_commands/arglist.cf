##############################################################################
#
# CFE-2724: Test that arglist attribute preserves white-spaces
#
##############################################################################

body common control
{
  inputs => { "../default.cf.sub" };
  bundlesequence  => { default("$(this.promise_filename)") };
  version => "1.0";
}

##############################################################################

bundle agent init
{
  files:
    "$(G.testfile).actual"
      delete => tidy;
    "$(G.testfile).sh"
      perms => mog("700", "root", "root"),
      content => '#!/bin/sh
for arg in "$(const.dollar)$(const.at)"
do
    echo "$(const.dollar)arg" >> $(G.testfile).actual
done';
}

##############################################################################

bundle agent test
{
  meta:
    "description" -> { "CFE-2724" }
      string => "Test that arglist attribute preserves white-spaces";
    "test_skip_unsupported"
      string => "windows",
      comment => "See ticket CFE-4294";

  commands:
    "$(G.testfile).sh"
      args => "one two three",
      arglist => { "four", "five six", " seven$(const.t)" };
}

##############################################################################

bundle agent check
{
  vars:
    "actual"
      string => readfile("$(G.testfile).actual");
    "expected"
      string => "one
two
three
four
five six
 seven$(const.t)
";

  methods:
    "any"
      usebundle => dcs_check_strcmp("$(actual)", "$(expected)",
                                    "$(this.promise_filename)", "no");
}
