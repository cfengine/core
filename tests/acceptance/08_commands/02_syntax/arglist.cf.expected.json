{
  "abc": "2'3\"4",
  "def": "567 =ghi=8'9\"a"
}
